test/README.md

```diff
@@ -1,95 +1,284 @@
+sudo apt-get install moby-engine
+~~~
+###### Create TLS certificates
+We will be running the container registry with TLS and authentication turned on. If you already have certs for TLS, feel free to use those and skip the cert creation steps listed below. 
+In a convenient directory, create a folder called `auth` to store your keys and certs (*I have used /home/azureuser/auth*)
+~~~sh
+mkdir auth
+cd auth
+# Create a private key
+openssl genrsa -out server.key  2048
+# Create a csr
+openssl req -new -key server.key -out server.csr
+# Create a cert from the csr
+openssl x509 -req -days 365 -in server.csr -signkey server.key -out server.crt
+
+# At the end of these steps, you have 3 files in your auth directory - server.key (the  
+# key), server.csr (the certificate signing request) and, server.crt (the certificate)
+~~~
+
+###### Setup native basic auth
+Next, we will setup basic username/password authentication for the registry using [htpasswd](http://httpd.apache.org/docs/current/programs/htpasswd.html).
+~~~ sh
+sudo apt-get install apache2-utils
+# Create password file
+: | sudo tee htpasswd
+# Create username and password. Replace the username and password strings 
+# below with a user/password combo that you like
+echo "password" | sudo htpasswd -iB htpasswd username
+~~~
+
+###### Configure and run the local container repository
+~~~ sh
+# Set up the container registry to listen on port 5000 and to restart automatically.
+# Mount the auth folder into the container and point to the TLS and Basic auth files
+# Restart is set to always - This will ensure that the registry is always up.
+sudo docker run -d  -p 5000:5000 \
+  --name registry  \
+  -v /home/azureuser/auth/:/etc/security \
+  -e REGISTRY_HTTP_TLS_CERTIFICATE=/etc/security/server.crt  \
+  -e REGISTRY_HTTP_TLS_KEY=/etc/security/server.key \
+  -e REGISTRY_AUTH=htpasswd   \
+  -e REGISTRY_AUTH_HTPASSWD_PATH=/etc/security/htpasswd   \
+  -e REGISTRY_AUTH_HTPASSWD_REALM="Registry Realm"  \
+  --restart always  \
+  registry:2
+
+# Run docker ps to validate that the registry came up successfully
+# If all goes well, you will see the something like below when you run docker ps.
+# 59867fe47a5b   registry:2   "/entrypoint.sh /etc…"   6 seconds ago   Up 6 seconds   0.0.0.0:5000->5000/tcp, :::5000->5000/tcp   registry
+sudo docker ps
+
+# Run docker login to validate your setup. The login step will cache your credentials on 
+# your machine and will be required later when you build and push docker images
+
+sudo docker login
+~~~
+
+##### Install .NET
+The end-to-end tests are written in .NET Core and run with the `dotnet test` command, and you need to install .NET Core SDK. See [Install the .NET SDK](https://docs.microsoft.com/en-us/dotnet/core/install/sdk) for further information. *Note: You will have to install .NET version 3.1.*
+
+Here is a convenient way to install the SDK.
+~~~ sh
+curl -sL https://dot.net/v1/dotnet-install.sh -o dotnet-install.sh
+./dotnet-install.sh -c Current

```

Comment:
I had to run `chmod u+x dotnet-install.sh` before this command.

Reply:
I recall I had to do something similar. Will add.